### PR TITLE
OHRM5X-851: Menu enhancement added for index pages

### PR DESCRIPTION
**Pull request by VageeshOrangeHRM**
This PR Contains following changes:
1. Module enhancement added for index pages.
2. Module under development missed URLs added.
3. Dao test added for newly added methods.

### CODE REVIEW

File: symfony/plugins/orangehrmCorePlugin/Dao/MenuDao.php [hunk 1]
```diff
@@ -118,4 +119,32 @@ public function enableModuleMenuItems(string $moduleName, array $menuTitles = []
...
         }
     }
+
+    /**
+     * @param string $screenName
+     * @return MenuItem|null
+     */
+    public function getMenuLevel(string $screenName): ?MenuItem
+    {
+        $screen = $this->getScreenByActionUrl($screenName);
```
Comment (RajithaKumara) [inline symfony/plugins/orangehrmCorePlugin/Dao/MenuDao.php:129]: why we need to fetch like this?

File: symfony/plugins/orangehrmCorePlugin/Dao/MenuDao.php [hunk 2]
```diff
@@ -118,4 +119,32 @@ public function enableModuleMenuItems(string $moduleName, array $menuTitles = []
...
+        $q = $this->createQueryBuilder(MenuItem::class, 'mi');
+        $q->leftJoin('mi.screen', 'sc');
+        $q->andWhere('sc.actionUrl = :screenName');
+        $q->setParameter('screenName', $screenName);
+        if ($screen) {
+            $q->andWhere('sc.id = :screenId');
+            $q->setParameter('screenId', $screen->getId());
+        }
```
Comment (RajithaKumara) [inline symfony/plugins/orangehrmCorePlugin/Dao/MenuDao.php:137]: what happen if we remove these lines

File: symfony/plugins/orangehrmCorePlugin/Dao/MenuDao.php [hunk 3]
```diff
@@ -118,4 +119,32 @@ public function enableModuleMenuItems(string $moduleName, array $menuTitles = []
             throw new DaoException($e->getMessage(), $e->getCode(), $e);
         }
     }
+
+    /**
+     * @param string $screenName
+     * @return MenuItem|null
+     */
+    public function getMenuLevel(string $screenName): ?MenuItem
```
Comment (RajithaKumara) [inline symfony/plugins/orangehrmCorePlugin/Dao/MenuDao.php:127]:
```php
public function getMenuLevel(string $module, string $screenName)
```
and method name should change. e.g. `getMenuItemByModuleAndScreen`

File: symfony/plugins/orangehrmCorePlugin/test/Dao/MenuDaoTest.php [hunk 4]
```diff
@@ -187,4 +187,14 @@ public function testEnableModuleMenuItemsByTitle()
         $count = $statement->fetchOne();
         $this->assertEquals(0, $count);
     }
+
+    public function testGetMenuLevel()
```
Comment (RajithaKumara) [inline symfony/plugins/orangehrmCorePlugin/test/Dao/MenuDaoTest.php:191]:
```php
public function testGetMenuLevel(): void
```

**Review by RajithaKumara — CHANGES_REQUESTED** [review]

File: symfony/plugins/orangehrmCorePlugin/test/Dao/MenuDaoTest.php (hunk 4 above)
Comment (VageeshOrangeHRM) [reply symfony/plugins/orangehrmCorePlugin/test/Dao/MenuDaoTest.php:191]: done, on that file, :void not added for the function, so i followed that way.

File: symfony/plugins/orangehrmCorePlugin/Dao/MenuDao.php (hunk 3 above)
Comment (VageeshOrangeHRM) [reply symfony/plugins/orangehrmCorePlugin/Dao/MenuDao.php:127]: done

File: symfony/plugins/orangehrmCorePlugin/Dao/MenuDao.php (hunk 1 above)
Comment (VageeshOrangeHRM) [reply symfony/plugins/orangehrmCorePlugin/Dao/MenuDao.php]: done

File: symfony/plugins/orangehrmCorePlugin/Dao/MenuDao.php (hunk 2 above)
Comment (VageeshOrangeHRM) [reply symfony/plugins/orangehrmCorePlugin/Dao/MenuDao.php]: done

**Pull request merged**